FWCore/Framework/src/EventProcessor.cc

```diff
@@ -1159,29 +1168,32 @@ namespace edm {
                   beginStreamTransitionAsync<Traits>(WaitingTaskHolder{eventTask},
                                                      *schedule_,i,*lp,ts,es,
                                                      serviceToken_,subProcesses_);
+                  status.reset();
                 });
               }
+              status.reset();
             }
-          });
-          
+          });  // beginStreamTask
+
           //task to start the global begin lumi
           WaitingTaskHolder beginStreamsHolder{beginStreamsTask};
           auto const& es = esp_->eventSetup();
           {
             typedef OccurrenceTraits<LuminosityBlockPrincipal, BranchActionGlobalBegin> Traits;
             beginGlobalTransitionAsync<Traits>(beginStreamsHolder,
                                                *schedule_,
-                                               *(status->lumiPrincipal()),
+                                               lumiPrincipal,
```

Comment:
This line is just a clean up where an already existing local variable is used instead.

Reply:
It is more than that. Because I moved status into beginStreamTask, it is already null at this point.